src/portable/config.rs

```diff
@@ -98,52 +106,69 @@ pub fn read(path: &Path) -> anyhow::Result<Config> {
     })
 }
 
-fn toml_set_version(data: &str, version: &Query)
-    -> anyhow::Result<Option<String>>
+pub fn modify_core<T, U, V, W>(parsed: &W, input: &String, config: &Path, selector: T, value: &U, format: V) -> anyhow::Result<bool>
+    where
+        T: Fn(&W) -> &Option<Spanned<U>>,
+        U: std::cmp::PartialEq,
+        V: FnOnce(&U) -> String,
```

Comment:
This is a lot of generics. Are all of them really needed?

If yes, please give them with more readable names i.e. (`Selector`).